src/Zabbix/Sender.php

```diff
@@ -132,10 +141,24 @@ protected function buildRequestBody()
     protected function sendData($body)
     {
         $socket = socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
+        if (!$socket) {
+            throw new \RuntimeException(socket_strerror(socket_last_error()));
+        }
+
+        // Set send and receive timeout
+        $timeoutSettings = ['sec' => floor($this->timeout), 'usec' => ($this->timeout - floor($this->timeout)) * 1e6];
```

Comment:
Apply timeout setting if timeout is not empty

Reply:
When SO_SNDTIMEO or SO_RCVTIMEO gets set to 0 it's the equivalent of infinite, which is the default behaviour. 
(https://linux.die.net/man/7/socket)
I've added some type casting in the constructor just to be explicit.